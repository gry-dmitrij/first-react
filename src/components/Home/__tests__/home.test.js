import {render} from "@testing-library/react";
import {Home} from "../index";
import {BrowserRouter} from "react-router-dom";

describe('Home component test', () => {
    it('snapshot test signup', () => {
        const signComponent = render(
            <BrowserRouter><Home onSignUp={() => null}/></BrowserRouter>);
        expect(signComponent).toMatchSnapshot();
    });
    it('snapshot test login', () => {
        const loginComponent = render(<BrowserRouter><Home onLogin={() => null}/></BrowserRouter>);
        expect(loginComponent).toMatchSnapshot();
    })
})
