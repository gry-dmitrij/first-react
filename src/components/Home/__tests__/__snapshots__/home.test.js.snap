// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component test snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1>
        Home
      </h1>
      <h2>
        Sign Up
      </h2>
      <a
        href="/login"
      >
        Login
      </a>
      <form>
        <label
          for="userName"
        >
          User name:
        </label>
        <input
          id="userName"
          type="text"
        />
        <br />
        <label
          for="userPass"
        >
          Password:
        </label>
        <input
          id="userPass"
          type="password"
        />
        <br />
        <input
          type="submit"
          value="Sign up"
        />
      </form>
    </div>
  </body>,
  "container": <div>
    <h1>
      Home
    </h1>
    <h2>
      Sign Up
    </h2>
    <a
      href="/login"
    >
      Login
    </a>
    <form>
      <label
        for="userName"
      >
        User name:
      </label>
      <input
        id="userName"
        type="text"
      />
      <br />
      <label
        for="userPass"
      >
        Password:
      </label>
      <input
        id="userPass"
        type="password"
      />
      <br />
      <input
        type="submit"
        value="Sign up"
      />
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home component test snapshot test 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1>
        Home
      </h1>
      <h2>
        Sign Up
      </h2>
      <a
        href="/login"
      >
        Login
      </a>
      <form>
        <label
          for="userName"
        >
          User name:
        </label>
        <input
          id="userName"
          type="text"
        />
        <br />
        <label
          for="userPass"
        >
          Password:
        </label>
        <input
          id="userPass"
          type="password"
        />
        <br />
        <input
          type="submit"
          value="Sign up"
        />
      </form>
    </div>
    <div>
      <h1>
        Home
      </h1>
      <h2>
        Login
      </h2>
      <a
        href="/signup"
      >
        Sign up
      </a>
      <form>
        <label
          for="userName"
        >
          User name:
        </label>
        <input
          id="userName"
          type="text"
        />
        <br />
        <label
          for="userPass"
        >
          Password:
        </label>
        <input
          id="userPass"
          type="password"
        />
        <br />
        <input
          type="submit"
          value="Login"
        />
      </form>
    </div>
  </body>,
  "container": <div>
    <h1>
      Home
    </h1>
    <h2>
      Login
    </h2>
    <a
      href="/signup"
    >
      Sign up
    </a>
    <form>
      <label
        for="userName"
      >
        User name:
      </label>
      <input
        id="userName"
        type="text"
      />
      <br />
      <label
        for="userPass"
      >
        Password:
      </label>
      <input
        id="userPass"
        type="password"
      />
      <br />
      <input
        type="submit"
        value="Login"
      />
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home component test snapshot test login 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1>
        Home
      </h1>
      <h2>
        Login
      </h2>
      <a
        href="/signup"
      >
        Sign up
      </a>
      <form>
        <label
          for="userName"
        >
          User name:
        </label>
        <input
          id="userName"
          type="text"
        />
        <br />
        <label
          for="userPass"
        >
          Password:
        </label>
        <input
          id="userPass"
          type="password"
        />
        <br />
        <input
          type="submit"
          value="Login"
        />
      </form>
    </div>
  </body>,
  "container": <div>
    <h1>
      Home
    </h1>
    <h2>
      Login
    </h2>
    <a
      href="/signup"
    >
      Sign up
    </a>
    <form>
      <label
        for="userName"
      >
        User name:
      </label>
      <input
        id="userName"
        type="text"
      />
      <br />
      <label
        for="userPass"
      >
        Password:
      </label>
      <input
        id="userPass"
        type="password"
      />
      <br />
      <input
        type="submit"
        value="Login"
      />
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home component test snapshot test signup 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1>
        Home
      </h1>
      <h2>
        Sign Up
      </h2>
      <a
        href="/login"
      >
        Login
      </a>
      <form>
        <label
          for="userName"
        >
          User name:
        </label>
        <input
          id="userName"
          type="text"
        />
        <br />
        <label
          for="userPass"
        >
          Password:
        </label>
        <input
          id="userPass"
          type="password"
        />
        <br />
        <input
          type="submit"
          value="Sign up"
        />
      </form>
    </div>
  </body>,
  "container": <div>
    <h1>
      Home
    </h1>
    <h2>
      Sign Up
    </h2>
    <a
      href="/login"
    >
      Login
    </a>
    <form>
      <label
        for="userName"
      >
        User name:
      </label>
      <input
        id="userName"
        type="text"
      />
      <br />
      <label
        for="userPass"
      >
        Password:
      </label>
      <input
        id="userPass"
        type="password"
      />
      <br />
      <input
        type="submit"
        value="Sign up"
      />
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
